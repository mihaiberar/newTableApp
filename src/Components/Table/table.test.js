import { LoadButton } from '../LoadButton/loadButton.jsx'
import { Table } from '../Table/table.jsx'
import { Context } from '../Context/context'
import { createMemoryHistory } from 'history'

import { render } from '@testing-library/react'
import { Router } from 'react-router'

it('should capture <Table /> snapshot', () => {
  const history = createMemoryHistory()
  const data = [{ name: 'Mihai' }]
  const { container } = render(
    <Router history={history}>
      <Context.Provider value={[data]}>
        <Table />
      </Context.Provider>
    </Router>
  )
  expect(container).toMatchSnapshot()
})

it('should render <LoadButton/> 5 times', () => {
  const history = createMemoryHistory()
  const data = [{ name: 'Mihai' }]
  const { container } = render(
    <Router history={history}>
      <Context.Provider value={[data]}>
        <Table />
      </Context.Provider>
    </Router>
  )
  expect(container.contains(LoadButton)).toHaveLength(5)
})
