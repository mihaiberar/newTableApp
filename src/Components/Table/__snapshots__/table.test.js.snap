// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should capture <Table /> snapshot 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <input
        class="searchInput"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="clearBtn"
      >
        X
      </button>
      <p
        style="color: white;"
      >
        0 rows selected
      </p>
      <button
        class="loadBtn"
        style="color: black;"
      >
        Load 5 rows
      </button>
      <button
        class="loadBtn"
        style="color: black;"
      >
        Load 10 rows
      </button>
      <button
        class="loadBtn"
        style="color: black;"
      >
        Load 20 rows
      </button>
      <button
        class="loadBtn"
        style="color: black;"
      >
        Load 50 rows
      </button>
      <button
        class="loadBtn"
        style="color: black;"
      >
        Load 100 rows
      </button>
      <button
        class="removeRowsBtn"
      >
        Remove rows
      </button>
    </div>
    <div
      class="table"
    >
      <div
        class="tableRow tableHeader"
      >
        <div
          class="tableCell"
        >
          <input
            class="inputCheckBox"
            readonly=""
            type="checkbox"
          />
        </div>
        <div
          class="tableCell"
        >
          Edit
        </div>
        <div
          class="tableCell"
        >
          Created at
        </div>
        <div
          class="tableCell"
        >
          ID
        </div>
        <div
          class="tableCell"
        >
          Name
        </div>
        <div
          class="tableCell"
        >
          Owner
        </div>
        <div
          class="tableCell"
        >
          Updated at
        </div>
        <div
          class="tableCell"
        >
          URL
        </div>
      </div>
      <div
        class="tableRow"
        style="background-color: rgb(114, 155, 156);"
      >
        <div
          class="tableCell"
        >
          <input
            class="inputCheckBox"
            readonly=""
            type="checkbox"
          />
        </div>
        <div
          class="tableCell"
        >
          <a
            href="/edit/undefined"
            style="color: black; text-decoration: none;"
          >
            <p>
              Edit
            </p>
          </a>
        </div>
        <div
          class="tableCell"
        >
          <span />
        </div>
        <div
          class="tableCell"
        >
          <span />
        </div>
        <div
          class="tableCell"
        >
          <span>
            Mihai
          </span>
        </div>
        <div
          class="tableCell"
        >
          <span />
        </div>
        <div
          class="tableCell"
        >
          <span />
        </div>
        <div
          class="tableCell"
        >
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
